src/cargo/ops/cargo_test.rs

```diff
@@ -114,15 +119,20 @@ fn run_doc_tests(options: &TestOptions,
                  -> CargoResult<Vec<ProcessError>> {
     let mut errors = Vec::new();
     let config = options.compile_opts.config;
-    let libs = compilation.package.targets().iter()
-                    .filter(|t| t.doctested())
-                    .map(|t| (t.src_path(), t.name(), t.crate_name()));
-    for (lib, name, crate_name) in libs {
+
+    let mut libs = vec![];
```

Comment:
Thanks, I've changed the code. I did not manage to do it with `flat_map`, because the borrow checker complained that `package` does not live long enough (when used in `map`):

     let libs = compilation.to_doc_test.iter().flat_map(|package| {
         package.targets().iter().filter(|t| t.doctested())
                          .map(|t| (package, t.src_path(), t.name(), t.crate_name()))
     }).collect::<Vec<_>>();

Now libs is an iterator over `(&Package, Iterator<(lib, name, create_name))`